src/test/java/com/yelp/nrtsearch/server/luceneserver/field/VectorFieldDefTest.java

```diff
@@ -0,0 +1,126 @@
+        FIELD_NAME, AddDocumentRequest.MultiValuedField.newBuilder().addValue(value).build());
+    return fieldsMap;
+  }
+
+  private List<AddDocumentRequest> buildDocuments(
+      String indexName, List<String> vectorFieldValues) {
+    List<AddDocumentRequest> documentRequests = new ArrayList<>();
+    for (String value : vectorFieldValues) {
+      documentRequests.add(
+          AddDocumentRequest.newBuilder()
+              .setIndexName(indexName)
+              .putAllFields(getFieldsMapForOneDocument(value))
+              .build());
+    }
+    return documentRequests;
+  }
+
+  @Override
+  public FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/field/registerFieldsVector.json");
+  }
+
+  @Override
+  public void initIndex(String name) throws Exception {
+    List<AddDocumentRequest> documents = buildDocuments(name, VECTOR_FIELD_VALUES);
+    addDocuments(documents.stream());
+  }
+
+  public FieldDef getFieldDef(String testIndex, String fieldName) throws IOException {
+    FieldDef fieldDef = getGrpcServer().getGlobalState().getIndex(testIndex).getField(fieldName);
+    return fieldDef;
+  }
+
+  @Test
+  public void vectorFieldDefTest() throws IOException {
+    FieldDef vectorFieldDef = getFieldDef(DEFAULT_TEST_INDEX, FIELD_NAME);
+    assertEquals(FIELD_TYPE, vectorFieldDef.getType());
+    assertEquals(FIELD_NAME, vectorFieldDef.getName());
+  }
+
+  @Test
+  public void vectorFieldDefDimensionMismatchTest() {
+    List<String> vectorFields = Arrays.asList("[1.0, 2.5]");
+    List<AddDocumentRequest> documents = buildDocuments(DEFAULT_TEST_INDEX, vectorFields);
+    Exception exception =
+        Assert.assertThrows(RuntimeException.class, () -> addDocuments(documents.stream()));
+    assertTrue(
+        exception
+            .getMessage()
+            .contains("The size of the vector data should match vectorDimensions field property"));
+  }
+
+  @Test
+  public void vectorFieldDefMultiValueExceptionTest() {
+    List<AddDocumentRequest> documentRequests = new ArrayList<>();
+    documentRequests.add(
+        AddDocumentRequest.newBuilder()
+            .setIndexName(DEFAULT_TEST_INDEX)
+            .putFields(
+                FIELD_NAME,
+                AddDocumentRequest.MultiValuedField.newBuilder()
+                    .addAllValue(VECTOR_FIELD_VALUES)
+                    .build())
+            .build());
+    Exception exception =
+        Assert.assertThrows(RuntimeException.class, () -> addDocuments(documentRequests.stream()));
+    assertTrue(
+        exception.getMessage().contains("Cannot index multiple values into single value field"));
+  }
+
+  @Test
+  public void parseVectorFieldToFloatArrTest() {
+    float[] expected = {1.0f, 2.5f, 1000.1000f};
+    String testJson = "[1.0, 2.5, 1000.1000]";
+
+    float[] resultVector = VectorFieldDef.parseVectorFieldToFloatArr(testJson);
+    assertTrue(Arrays.equals(expected, resultVector));
+  }
+}
```

Comment:
Might be good to have a test that does a search query that retrieves the vector field, and verifies it is present in the search response.